h/schemas/forms/admin/group.py

```diff
@@ -30,15 +30,24 @@ def username_validator(form, value):
     group and the group's organization.
     """
     # Unlike other validators, this one is applied at the root level of the
-    # form. This us because we need to read the value from the organisation
+    # form. This is because we need to read the value from the organisation
     # to get the right authority to check the users. This isn't possible in
     # a specific validator, as it has no access to the global form data.
 
     user_svc = form.bindings["user_svc"]
 
-    # Get the authority from the list of organisations based on the one
-    # actually picked in the form
-    authority = form.bindings["organizations"][value["organization"]].authority
+    if group := form.bindings.get("group"):
+        authority = group.authority
+    elif organization := value.get("organization"):
+        # If there's no group then fall back on the organization's authority
+        # (this happens when creating a new group as opposed to editing an
+        # existing one).
+        authority = form.bindings.get("organizations")[organization].authority
+    else:
+        # If there's no group or organisation then fall back on the default authority
+        # (this happens when creating a new group and selecting no organization
+        # for the group to belong to).
+        authority = form.bindings["request"].default_authority
```

Comment:
The local variable `authority` is used below to validate that the group's creator and members (both of which the user can edit in the admin form) all belong to the same authority as the group. But `authority` has to be derived differently in different cases. There are 4 cases here:

1. When editing an existing group that does have an organization: in this case `group.authority` will be used
2. When editing an existing group that does _not_ have an organization: again `group.authority` will be used
3. When creating a new group with an organization: `organization.authority` will be used
4. When creating a new group _without_ an organization: `request.default_authority` will be used